apps/settings/js/modules/addon_manager.js

```diff
@@ -43,7 +43,11 @@ define(function(require) {
      * @memberOf AddonManager.prototype
      */
     _bindEvents: function() {
-      AppsCache.addEventListener('oninstall', this._updateAddons.bind(this));
+      AppsCache.addEventListener('oninstall', (evt) => {
+        evt.application.addEventListener('downloadsuccess', (evt) => {
+          this._updateAddons(evt);
+        });
+      });
       AppsCache.addEventListener('onuninstall', this._updateAddons.bind(this));
```

Comment:
mmm, seems current code is not ideal, we could create a _boundUpdateAddons object to bind _updateAddons and reuse the same object to handle these 3 events